### PR TITLE
Creative-mode timer

### PR DESCRIPTION
**Pull request by rjawesome**
https://github.com/biothings/biothings_explorer/issues/808
Switches result-limit in creative mode for a limit based on estimated completion time of templates.

### CODE REVIEW

File: src/inferred_mode/inferred_mode.ts [hunk 1]
```diff
@@ -610,11 +586,7 @@ export default class InferredQueryHandler {
         new LogEntry(
           'INFO',
           null,
-          [
-            `Final result count`,
-            Object.keys(combinedResponse.message.results).length > this.CREATIVE_LIMIT ? ' (before truncation):' : ':',
-            ` ${Object.keys(combinedResponse.message.results).length}`,
-          ].join(''),
+          `Final result count: ${Object.keys(combinedResponse.message.results).length}`,
```
Comment (tokebe) [inline src/inferred_mode/inferred_mode.ts:589]: We still want to log how many results before/after truncation

File: src/inferred_mode/inferred_mode.ts [hunk 2]
```diff
@@ -623,8 +595,7 @@ export default class InferredQueryHandler {
     response.message.results = Object.values(combinedResponse.message.results).sort((a, b) => {
       return b.analyses[0].score - a.analyses[0].score ? b.analyses[0].score - a.analyses[0].score : 0;
     });
-    // trim extra results and prune kg
-    response.message.results = response.message.results.slice(0, this.CREATIVE_LIMIT);
```
Comment (tokebe) [inline src/inferred_mode/inferred_mode.ts:617]: Results should still be truncated to the top 500 at the end per Translator expectations.

File: src/inferred_mode/inferred_mode.ts [hunk 3]
```diff
@@ -69,7 +67,6 @@ export default class InferredQueryHandler {
     this.path = path;
     this.predicatePath = predicatePath;
     this.includeReasoner = includeReasoner;
-    this.CREATIVE_LIMIT = process.env.CREATIVE_LIMIT ? parseInt(process.env.CREATIVE_LIMIT) : 500;
```
Comment (tokebe) [inline src/inferred_mode/inferred_mode.ts:72]: Creative limit should be kept, but used only for truncation, rather than as a stopping threshold

**Review by tokebe — CHANGES_REQUESTED** [review]
Made a few notes on desired behavior, please review and make the appropriate changes.

**Comment by rjawesome** [comment]
Creative limit results truncation has been added back.

**Comment by tokebe** [comment]
Closing as parent issue is superseded

**Pull request closed without merging**